app/src/ui/cherry-pick/choose-target-branch.tsx

```diff
@@ -0,0 +1,173 @@
+    this.setState({ filterText })
+  }
+
+  private onSelectionChanged = (selectedBranch: Branch | null) => {
+    this.setState({ selectedBranch })
+  }
+
+  private renderBranch = (item: IBranchListItem, matches: IMatches) => {
+    return renderDefaultBranch(item, matches, this.props.currentBranch)
+  }
+
+  private onEnterPressed = (branch: Branch, source: ClickSource) => {
+    if (source.kind !== 'keyboard' || source.event.key !== 'Enter') {
+      return
+    }
+
+    source.event.preventDefault()
+
+    const { selectedBranch } = this.state
+
+    if (selectedBranch !== null && selectedBranch.name === branch.name) {
+      this.startCherryPick()
+    }
+  }
+
+  private canCherryPickOntoSelectedBranch() {
+    const { selectedBranch } = this.state
+    return selectedBranch !== null && !this.selectedBranchIsCurrentBranch()
+  }
+
+  private selectedBranchIsCurrentBranch() {
+    const { selectedBranch } = this.state
+    const currentBranch = this.props.currentBranch
+    return (
+      selectedBranch !== null &&
+      currentBranch !== null &&
+      selectedBranch.name === currentBranch.name
+    )
+  }
+
+  private renderOkButtonText() {
+    const okButtonText = 'Cherry pick commit'
+
+    const { selectedBranch } = this.state
+    if (selectedBranch !== null) {
+      return (
+        <>
+          {okButtonText} to <strong>{selectedBranch.name}</strong>…
+        </>
+      )
+    }
+
+    return okButtonText
+  }
+
+  public render() {
+    const tooltip = this.selectedBranchIsCurrentBranch()
+      ? 'You are not able to cherry pick from and to the same branch'
+      : undefined
+
+    return (
+      <Dialog
+        id="cherry-pick"
+        onDismissed={this.props.onDismissed}
+        onSubmit={this.startCherryPick}
+        dismissable={true}
+        title={<strong>Cherry pick commit to a branch</strong>}
+      >
+        <DialogContent>
+          <BranchList
+            allBranches={this.props.allBranches}
+            currentBranch={this.props.currentBranch}
+            defaultBranch={this.props.defaultBranch}
+            recentBranches={this.props.recentBranches}
+            filterText={this.state.filterText}
+            onFilterTextChanged={this.onFilterTextChanged}
+            selectedBranch={this.state.selectedBranch}
+            onSelectionChanged={this.onSelectionChanged}
+            canCreateNewBranch={false}
+            renderBranch={this.renderBranch}
+            onItemClick={this.onEnterPressed}
+          />
+        </DialogContent>
+        <DialogFooter>
+          <OkCancelButtonGroup
+            okButtonText={this.renderOkButtonText()}
+            okButtonDisabled={!this.canCherryPickOntoSelectedBranch()}
+            okButtonTitle={tooltip}
+            cancelButtonVisible={false}
+          />
+        </DialogFooter>
+      </Dialog>
+    )
+  }
+
+  private startCherryPick = async () => {
+    const { selectedBranch } = this.state
+    if (!selectedBranch) {
```

Comment:
I think it's preferred to use `!== null` explicitly (at least that's what Markus told me a few times 🤣 )
```suggestion
    if (selectedBranch !== null) {
```

Reply:
Absolutely! I blame habit. 😸  Markus called me out on that too. 🙂  Thank you. Actually now that you point it out, I made this method before I made the "canCherryPickOntoSelectedBranch" method which would be a more complete failsafe here.